packages/web3-core-method/tests/src/methods/node/GetProofMethodTest.js

```diff
@@ -0,0 +1,69 @@
+import * as Utils from 'web3-utils';
+import AbstractMethod from '../../../../lib/methods/AbstractMethod';
+import GetProofMethod from '../../../../src/methods/node/GetProofMethod';
+
+/**
+ * GetProofMethod test
+ */
+describe('GetProofMethodTest', () => {
+    let method;
+
+    beforeEach(() => {
+        method = new GetProofMethod(Utils, null, {});
+    });
+
+    it('constructor check', () => {
+        expect(method).toBeInstanceOf(AbstractMethod);
+
+        expect(method.rpcMethod).toEqual('eth_getProof');
+
+        expect(method.parametersAmount).toEqual(3);
+
+        expect(method.utils).toEqual(Utils);
+
+        expect(method.formatters).toEqual(null);
+    });
+
+    it('afterExecution should map the response', () => {
+        const response = {
+            jsonrpc: '2.0',
+            id: 1,
+            result: {
```

Comment:
the ``afterExecution`` method just get the result object and not the complete JSON-RPC response.

Reply:
Sorry my mistake.

Reply:
```
> eth.getProof("0x651c77c61b4a53a3c1d2e062796770af5feb9380", INFO [04-24|15:59:57.894] New local node record                    seq=198 id=920485397a8116e7 ip=14.143.40.34 udp=30303 tcp=30303
> eth.getProof("0x651c77c61b4a53a3c1d2e062796770af5feb9380", ["0x0000000000000000000000000000000000000000000000000000000000000000","0x0000000000000000000000000000000000000000000000000000000000000001"],"latest")
{
  accountProof: ["0xf8518080a0517337efc973654f74640b6ef0529e88635e69b6e3731fc051281e33719c23a8a0f5482f368e10755e10c917e5370419a455885fc66901a33cdfbd7a510d722f5280808080808080808080808080", "0xf872a035e21f3f1ec4126dd0768297052fc8c04c64b8073461ca1ba04b7504824e2341b84ff84d0189013f306a2409fc0000a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"],
  address: "0x651c77c61b4a53a3c1d2e062796770af5feb9380",
  balance: "0x13f306a2409fc0000",
  codeHash: "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470",
  nonce: "0x1",
  storageHash: "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
  storageProof: [{
      key: "0x0000000000000000000000000000000000000000000000000000000000000000",
      proof: [],
      value: "0x0"
  }, {
      key: "0x0000000000000000000000000000000000000000000000000000000000000001",
      proof: [],
      value: "0x0"
  }]
}
```
Balance can also be a big number. I guess balance should also be converted to BN. @nivida.
I tested this in my local geth.

Reply:
Yes, balance should be a BN too. You could also transform the nonce because it's possible that the nonce will exceed the max integer value of JS.

Reply:
Also, should I convert balance from Wei to ether or just pass as it is?